### ISSUE TITLE
mozdownload@1.19 throws errors on OSX

### ISSUE THREAD

**Issue by pdehaan** [mozilla/mozdownload#345]
We were trying to use **mozdownload** today to download some Moz's and were getting the following error on OSX (El Capitan) using v1.19.

Behold:

```
$ mozdownload --version=latest
Traceback (most recent call last):
  File "/usr/local/bin/mozdownload", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python2.7/site-packages/mozdownload/cli.py", line 168, in cli
    build.download()
  File "/usr/local/lib/python2.7/site-packages/mozdownload/scraper.py", line 270, in download
    if os.path.isfile(os.path.abspath(self.filename)):
  File "/usr/local/lib/python2.7/site-packages/mozdownload/scraper.py", line 230, in filename
    self.build_filename(self.binary))
  File "/usr/local/lib/python2.7/site-packages/mozdownload/scraper.py", line 183, in binary
    raise errors.NotFoundError(message, self.path)
mozdownload.errors.NotFoundError: Specified build has not been found: https://archive.mozilla.org/pub/mozilla.org/firefox/releases/latest/mac/en-US/
```

```
$ mozdownload

Module to handle downloads for different types of archive.mozilla.org hosted
applications.

mozdownload version: 1.19

Specify --help for more information on args. Please see the README for examples.
```

**Comment by whimboo** [comment]
`latest` still exists on archive.mozilla.org but doesn't list any release anymore: http://archive.mozilla.org/pub/firefox/releases/latest/

So not sure if that folder is going to be dropped and not updated anymore. @nthomas-mozilla can help here.

**Comment by nthomas-mozilla** [comment]
We've discontinued it, please see
 http://archive.mozilla.org/pub/firefox/releases/latest/README.txt
 https://ftangftang.wordpress.com/2015/11/17/the-latest-on-firefoxreleaseslatest/

**Comment by whimboo** [comment]
I see. So if `latest` is still wanted we will have to reimplement the logic. Please let us know.

**Comment by whimboo** [comment]
This was fixed with my patch on issue #356.
